### PR TITLE
Support PostCSS 8

### PR DESCRIPTION
**Pull request by cl1ck**

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -1,36 +1,36 @@
 const assert = require('assert')
 const path = require('path')
-const plugin = require('./')
+const plugin = require('.\/')
```
Comment (giuseppeg) [inline test.js]: why this change?

File: test.js [hunk 2]
```diff
@@ -1,36 +1,36 @@
...
       plugin('p { color: %%styled-jsx-placeholder-0%%; & img { display: block; } } %%styled-jsx-placeholder-1%%')
       , 'p { color: %%styled-jsx-placeholder-0%% } p img { display: block; } %%styled-jsx-placeholder-1%%'
     )
   })
 
   it('works with @import', () => {
-    assert.equal(
-      plugin('@import "./fixture.css"; p { color: red }'),
+    assert.strictEqual(
+      plugin('@import ".\/fixture.css"; p { color: red }'),
```
Comment (giuseppeg) [inline test.js]: this shouldn't be escaped should it?

File: test.js [hunk 3]
```diff
@@ -44,7 +44,7 @@ describe('styled-jsx-plugin-postcss', () => {
       },
       {
         name: 'Error',
-        message: /postcss failed with CssSyntaxError: <css input>:2:12: Unclosed string/
+        message: /postcss failed with postcss.plugin was deprecated. Migration guide:\nhttps:\/\/evilmartians.com\/chronicles\/postcss-8-plugin-migration\nCssSyntaxError: <css input>:2:12: Unclosed string\n/
```
Comment (giuseppeg) [inline test.js]: this error is for when you are using an old API but the test is about checking that an error is thrown when trying to compile invalid css, can you revert this change and fix this (see where we use postcss.plugin)

File: test.js [hunk 4]
```diff
@@ -58,7 +58,7 @@ describe('styled-jsx-plugin-postcss', () => {
       },
       {
         name: 'Error',
-        message: /postcss failed with TypeError: Invalid PostCSS Plugin found: \[0\]/
+        message: /postcss failed with TypeError: Invalid PostCSS Plugin found at: plugins\[0]/
```
Comment (giuseppeg) [inline test.js:61]: same above

File: package.json [hunk 5]
```diff
@@ -1,10 +1,10 @@
 {
   "name": "styled-jsx-plugin-postcss",
-  "version": "3.0.2",
+  "version": "4.0.0",
   "description": "Plugin to add PostCSS support to styled-jsx",
   "main": "index.js",
   "scripts": {
-    "test": "mocha test.js"
+    "test": "mocha --timeout 5000 test.js"
```
Comment (giuseppeg) [inline package.json]: why this timeout?

**Review by giuseppeg — COMMENTED** [review]
Thanks for the PR and sorry for the delay. I don't have a lot of time to contribute to this project

File: test.js (hunk 1 above)
Comment (cl1ck) [reply test.js]: sorry, must have been a search / replace error - fixed it

File: test.js (hunk 2 above)
Comment (cl1ck) [reply test.js]: same... fixed

File: test.js (hunk 3 above)
Comment (cl1ck) [reply test.js]: Yeah, I got irritated by the wrong error message. I replaced the outdated test dependencies with PostCSS8 compatible ones and changed the test back to the old error message.

File: package.json (hunk 5 above)
Comment (cl1ck) [reply package.json]: The default timeout of mocha is 2 seconds which was sometimes too short.

File: test.js [hunk 6]
```diff
@@ -4,32 +4,32 @@ const plugin = require('./')
...
       plugin('p { color: %%styled-jsx-placeholder-0%%; & img { display: block; } } %%styled-jsx-placeholder-1%%')
       , 'p { color: %%styled-jsx-placeholder-0%% } p img { display: block; } %%styled-jsx-placeholder-1%%'
     )
   })
 
   it('works with @import', () => {
-    assert.equal(
-      plugin('@import "./fixture.css"; p { color: red }'),
+    assert.strictEqual(
+      plugin('@import ".\/fixture.css"; p { color: red }'),
```
Comment (giuseppeg) [inline test.js]:
```suggestion
      plugin('@import "./fixture.css"; p { color: red }'),
```

**Comment by giuseppeg** [comment]
thank you again Michel!
